src/platform_impl/windows/ime.rs

```diff
@@ -124,16 +124,24 @@ impl ImeContext {
             left: x,
             top: y,
             right: x + width,
-            bottom: y - height,
+            bottom: y + height,
         };
         let candidate_form = CANDIDATEFORM {
             dwIndex: 0,
             dwStyle: CFS_EXCLUDE,
             ptCurrentPos: POINT { x, y },
             rcArea: rc_area,
         };
+        let composition_form = COMPOSITIONFORM {
+            dwStyle: CFS_POINT,
+            ptCurrentPos: POINT { x, y: y + height },
+            rcArea: rc_area,
+        };
 
-        unsafe { ImmSetCandidateWindow(self.himc, &candidate_form) };
+        unsafe {
+            ImmSetCompositionWindow(self.himc, &composition_form);
+            ImmSetCandidateWindow(self.himc, &candidate_form);
```

Comment:
Why the order is like that and not other way around? Could we set after creating data for the request we want to perform and not doing just set-all in the end of function?

Also, if you want me to re-review you should request it.

Reply:
@kchibisov the order doesn't matter here. Create and set will be more nature, i have modify based on your comments